backend/src/ds/grpc.rs

```diff
@@ -280,7 +277,14 @@ impl<Qep: QsConnector> DeliveryService for GrpcDs<Qep> {
             .creator_client_reference
             .ok_or_missing_field("creator_client_reference")?
             .try_into()?;
-        let room_state = serde_json::from_slice(&payload.room_state).unwrap(); // TODO Handle error
+        let room_state = mimi_room_policy::RoomState::try_from_ref(
+            &payload.room_state.ok_or_missing_field("room_state")?,
+        )
+        .map_err(|_| Status::invalid_argument("Invalid room_state message"))?;
+
+        let room_state = VerifiedRoomState::verify(room_state)
+            .map_err(|_| Status::invalid_argument("Room state verification failed"))?;
```

Comment:
I would propose to log the error on the server, otherwise we would not know why it failed. The message for the client does not contain the error on purpose not to leak too much information.